### PR TITLE
use ssb-uri2 for better checks

### PR DESCRIPTION
**Pull request by staltz**

### CODE REVIEW

File: test/ssb-uri.test.js [hunk 1]
```diff
@@ -18,8 +21,5 @@ tape('ssb-uri encoding', function (t) {
     'unknown URIs get encoded as BFE strings'
   )
 
-  const weirdURI = 'ssb:feed/joke/invalidCanonicalSSBURI'
-  t.throws(() => bfe.encode(weirdURI), 'SSB URI with unknown format throws')
```
Comment (staltz) [inline test/ssb-uri.test.js:22]: Hmm, hold a minute. I think I can bring back this.

**Comment by staltz** [comment]
Ready for review

**Comment by mycognosist** [comment]
I'm taking a look at this now.

File: index.js [hunk 2]
```diff
@@ -44,7 +44,12 @@ const encoder = {
   },
 
   ssbURI(input) {
-    const [typeName, formatName, data = ''] = new URL(input).pathname.split('/')
+    // These URIs do not have BFE counterparts, so treat them as strings:
```
Comment (mycognosist) [inline index.js:47]: 👍🏻

File: index.js [hunk 3]
```diff
@@ -149,7 +154,8 @@ function encode(input) {
 const decoder = {
   ssbURI(input, type, format) {
     const d = input.slice(2)
-    return `ssb:${type.type}/${format.format}/${bufferToURIData(d)}`
+    const data = d.toString('base64')
+    return SSBURI.compose({ type: type.type, format: format.format, data })
```
Comment (mycognosist) [inline index.js:158]: Beautiful symmetry with `decompose` 🖤

**Review by mycognosist — APPROVED** [review]
Excellent. Great to have these stricter checks baked-in. Beautiful work as always :)

**Pull request merged**